llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py

```diff
@@ -1631,5 +1632,61 @@ def _create_query_result(
             nodes=node_result, similarities=score_result, ids=id_result
         )
 
+    async def _acreate_query_result(
+        self, search_query: str, vectors: Optional[List[Any]]
+    ) -> VectorStoreQueryResult:
+        results = await self._search_client.search(
```

Comment:
You might need to merge main, but this should be `self._async_search_client` -- I merged some changes to
a) better separate these clients
b) allow users to pass in both clients